elevation-plot/index.js

```diff
@@ -0,0 +1,109 @@
+import * as mapboxPolyline from '@mapbox/polyline';
+import { displaySpecs, updateSpecs } from './journey-specs';
+
+/**
+ * For the purpose of this example we use urgl - lightweights GraphQL client.
+ * To establish a connection with Chargetrip GraphQL API you need to have an API key.
+ * Read more about an authorisation in our documentation (https://docs.chargetrip.com/#authorisation).
+ */
+const headers = {
+  'x-client-id': '5e8c22366f9c5f23ab0eff39',
+};
+
+const subscriptionClient = new SubscriptionClient('wss://api.chargetrip.io/graphql', {
+  reconnect: true,
+  connectionParams: headers,
+});
+
+const client = createClient({
+  url: 'https://api.chargetrip.io/graphql',
+  fetchOptions: {
+    method: 'POST',
+    headers,
+  },
+  exchanges: [
+    ...defaultExchanges,
+    subscriptionExchange({
+      forwardSubscription(operation) {
+        return subscriptionClient.request(operation);
+      },
+    }),
+  ],
+});
+
+/**
+ * To create a route you need to:
+ *
+ * 1. create a new route and receive back its ID;
+ * 2. subscribe to route updates in order to receive its details.
+ */
+client
+  .mutation(createRoute)
+  .toPromise()
+  .then(response => {
+    const routeId = response.data.newRoute;
+
+    const { unsubscribe } = pipe(
+      client.executeSubscription(createRequest(routeUpdate, { id: routeId })),
+      subscribe(result => {
+        const { status, route } = result.data.routeUpdatedById;
+
+        // you can keep listening to the route changes to update route information
+        // for this example we want to only draw the initial route
+        if (status === 'done' && route) {
+          unsubscribe();
+          const routeData = result.data.routeUpdatedById.route;
+          drawRoutePolyline(routeData, response.data.newRoute);
+          loadGraph(routeData, routeData.elevationPlot);
+          imageLoader(routeData, routeData.legs);
+        }
+      }),
+    );
+  })
+  .catch(error => console.log(error));
+
+/**
+ * Draw a route on a map.
+ *
+ * Route object contains `polyline` data -  the polyline encoded route (series of coordinates as a single string).
+ * We need to decode this information first. We use Mapbox polyline tool (https://www.npmjs.com/package/@mapbox/polyline) for this.
+ * As a result of decoding we get pairs [latitude, longitude].
+ * To draw a route on a map we use Mapbox GL JS. This tool uses the format [longitude,latitude],
+ * so we have to reverse every pair.
+ *
+ * @param data {object} route specification
+ */
+const drawRoutePolyline = (data, id) => {
+  const decodedData = mapboxPolyline.decode(data.polyline);
+  const reversed = decodedData.map(item => item.reverse());
+
+  client
+    .query(getRoutePath(id, reversed[0]))
+    .toPromise()
+    .then(response => {
+      displaySpecs(response.data);
+    })
+    .catch(error => console.log(error));
+  const map = drawRoute(reversed, data.legs);
+  updateSpecs(client, map, reversed, id);
```

Comment:
It's not a very clean solution here (client and map as parameters, we should try to keep different functionalities separated and modular), I think we can find a better way, let's sit together ;)